### DISCUSSION TITLE
IdentityPartsSeparator does not work in test

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#14312] [Q&A] [answered]
I have the following test and I expect the Id to be `NIM-1` but the result is `NIM/1`? Is it not possible to change the separator in tests?

```c#
public class IdGenerationTest : RavenTestDriver
{
  protected override void PreInitialize(IDocumentStore documentStore)
  {
    documentStore.Conventions.IdentityPartsSeparator = '-';
  }

  [Fact]
  public void ShouldGenerateIdWithWorkspaceKeyAndIncrementor()
  {
    using (var store = GetDocumentStore())
    {
      using (var session = store.OpenSession())
      {
        var team = new Team
        {
          Name = "Team 1"
        };
        session.Store(team);
        session.SaveChanges();
      }

      using (var session = store.OpenSession())
      {
        var team = session.Query<Team>().First();
        Assert.NotNull(team);
        Console.WriteLine(team.Id);
      }

    }
  }
}

public class Team
{
  public string Id { get; set; }
  public string Name { get; set; }
  public Team()
  {
    Id = "NIM|";
  }
}
```

**Comment by omerlewitz** [comment, marked as answer]
Hi Marcus,

You are using `"MIN|"` in the `Id` property and this is _server-wide-generation_
In order to make it work you need to update the _client configuration_  using this _operation_ :

```
await store.Maintenance.SendAsync(new PutClientConfigurationOperation(new ClientConfiguration
{
    IdentityPartsSeparator = '-'
}));
```
Referring to the documentation - https://ravendb.net/docs/article-page/5.0/csharp/server/kb/document-identifier-generation

This is the `Identity` algorithm.

Alternatively, You can modify the _Seperator_ with the `Hilo` algorithm by _not initializing_ the `Id` and by _modifying_ the `DocumentStore` `Conventions`


**Test**

```
[Fact]
public async Task Marcus()
{
    var team = new Team();
    var databaseName = GetDatabaseName();
    using (var server = GetNewServer())
        
    using (var store = new DocumentStore
           {
               Database = databaseName,
               Urls = new[] {server.WebUrl}, 
           }.Initialize())
    {
        store.Maintenance.Server.Send(new CreateDatabaseOperation(new DatabaseRecord(databaseName)));
        await store.Maintenance.SendAsync(new PutClientConfigurationOperation(new ClientConfiguration
        {
            IdentityPartsSeparator = '-'
        }));
        using (var session = store.OpenSession())
        {
            session.Store(team);
            session.SaveChanges();
        }
    }
}
public class Team
{
    public string Id { get; set; }
    public string Name { get; set; }
    
    public Team()
    {
        Id = "NIM|";
    }
}
```

**Reply by marcuslindblom** [reply]
@omerlewitz Which HiLo convention should I configure to get the id looking like this `NIM-1`? If I remove `Id = "NIM|";` the Id ends up with `NIM-1-A`.

**Reply by ayende** [reply]
You need to make a distinction between _identity_ and the `/` in the hilo.
You can set the later by setting the `IdentityPartsSeparator` value in the DatabaseRecord.

**Reply by marcuslindblom** [reply]
@ayende Yes, setting `IdentityPartsSeparator` works but how do I remove the letter `A` from the id?

**Reply by marcuslindblom** [reply]
I guess the only way to remove `A` from the Id is to use cluster wide Id using `Id = "NIM|"`.  Using HiLo and modify the `IdentityPartsSeparator` like this 
```c#
await store.Maintenance.SendAsync(new PutClientConfigurationOperation(new ClientConfiguration
{
  IdentityPartsSeparator = '-'
}));
```
... adds the `A` because it's not cluster wide?

**Reply by ayende** [reply]
Hilo by its nature will add the node tag to the id, since you can have concurrent hilos from different nodes at the same time.
If you want to avoid that, you need to use the identity option, instead, yes.
